test/solidity/Periphery/LiFiDEXAggregator.t.sol

```diff
@@ -0,0 +1,320 @@
+                amountIn: 1_000 * 1e6,
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_Stable_Reverse() public {
+        // first perform the forward stable swap.
+        test_CanSwapViaVelodromeV2_Stable();
+
+        vm.startPrank(USER_SENDER);
+
+        _testSwap(
+            SwapTestParams({
+                from: USER_SENDER,
+                to: USER_SENDER,
+                tokenIn: address(USDC_E_TOKEN),
+                amountIn: 500 * 1e6,
+                tokenOut: address(USDC_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 1,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_FromDexAggregator() public {
+        // fund dex aggregator contract so that the contract holds USDC
+        deal(address(USDC_TOKEN), address(liFiDEXAggregator), 100_000 * 1e6);
+
+        vm.startPrank(USER_SENDER);
+        _testSwap(
+            SwapTestParams({
+                from: address(liFiDEXAggregator),
+                to: address(USER_SENDER),
+                tokenIn: address(USDC_TOKEN),
+                amountIn: USDC_TOKEN.balanceOf(address(liFiDEXAggregator)) - 1, // has to be current dex aggregator balance - 1
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: false
+            })
+        );
+        vm.stopPrank();
+    }
+
+    function test_CanSwapViaVelodromeV2_FlashloanCallback() public {
+        mockFlashloanCallbackReceiver = new MockVelodromeV2FlashLoanCallbackReceiver();
+
+        vm.startPrank(USER_SENDER);
+        _testSwap(
+            SwapTestParams({
+                from: address(USER_SENDER),
+                to: address(mockFlashloanCallbackReceiver),
+                tokenIn: address(USDC_TOKEN),
+                amountIn: 1_000 * 1e6,
+                tokenOut: address(USDC_E_TOKEN),
+                stable: true, // - NOT USED!
+                fee: 500, // - NOT USED!
+                direction: 0,
+                callback: true
+            })
+        );
+        vm.stopPrank();
+    }
+
+    struct SwapTestParams {
+        address from;
+        address to;
+        address tokenIn;
+        uint256 amountIn;
+        address tokenOut;
+        bool stable;
+        uint24 fee;
+        uint8 direction;
+        bool callback;
+    }
+
+    /**
+     * @dev Helper function to test a VelodromeV2 swap.
+     * Uses a struct to group parameters and reduce stack depth.
+     */
+    function _testSwap(SwapTestParams memory params) internal {
```

Comment:
very nicely coded (and commented 😀)  !!!

Reply:
thank you!